packages/contractkit/src/identity/offchain/storage-writers.ts

```diff
@@ -41,12 +41,32 @@ export class GitStorageWriter extends LocalStorageWriter {
   }
 }
 
+export class GoogleStorageWriter extends LocalStorageWriter {
+  private readonly bucket: string
+
+  constructor(readonly local: string, bucket: string) {
+    super(local)
+    this.bucket = bucket
+  }
+
+  async write(data: Buffer, dataPath: string): Promise<void> {
+    await this.writeToFs(data, dataPath)
+    execSync(`gsutil cp ${normalize(this.root + '/' + dataPath)} gs://${this.bucket}${dataPath}`, {
+      cwd: this.root,
+    })
+    await new Promise((resolve) => setTimeout(resolve, 2000))
+  }
+}
+
 export class MockStorageWriter extends LocalStorageWriter {
   constructor(readonly root: string, readonly mockedStorageRoot: string, readonly fetchMock: any) {
     super(root)
   }
-  async write(data: string, dataPath: string): Promise<void> {
+  async write(data: Buffer, dataPath: string): Promise<void> {
     await this.writeToFs(data, dataPath)
-    this.fetchMock.mock(this.mockedStorageRoot + dataPath, data)
+    this.fetchMock.mock(this.mockedStorageRoot + dataPath, data, {
```

Comment:
You might want to use `url.resolve` here instead of simple concatenation. Obviously it may not matter as much because its in a test, but still removes a sharp edge.